forui/lib/src/widgets/button/button_spinner.dart

```diff
@@ -41,38 +41,35 @@ class FButtonSpinner extends StatefulWidget {
 class _FButtonSpinnerState extends State<FButtonSpinner> with SingleTickerProviderStateMixin {
   late AnimationController _controller;
   late Animation<double> _animation;
+  late final FButtonSpinnerStyle _style;
 
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
-    final FButtonData(:style) = FButtonData.of(context);
+    _style = widget.style ?? FButtonData.of(context).style.spinnerStyle;
     _controller = AnimationController(
       vsync: this,
-      duration: widget.style?.animationDuration ?? style.spinnerStyle.animationDuration,
+      duration: _style.animationDuration,
     )
       ..forward()
       ..repeat();
     _animation = Tween<double>(begin: 0.0, end: 1.0).animate(_controller);
   }
 
   @override
-  Widget build(BuildContext context) {
-    final data = FButtonData.of(context);
-    final style = widget.style ?? data.style.spinnerStyle;
-    return AnimatedBuilder(
-      animation: _animation,
-      builder: (_, child) => Transform.rotate(
-        angle: _controller.value * 2 * math.pi,
-        child: child,
-      ),
-      child: FIcon(
-        FAssets.icons.loaderCircle,
-        color: data.enabled ? style.enabledSpinnerColor : style.enabledSpinnerColor,
-        size: style.spinnerSize,
-        semanticLabel: 'Button Spinner',
-      ),
-    );
-  }
+  Widget build(BuildContext context) => AnimatedBuilder(
+        animation: _animation,
+        builder: (_, child) => Transform.rotate(
+          angle: _controller.value * 2 * math.pi,
+          child: child,
+        ),
+        child: FIcon(
+          FAssets.icons.loaderCircle,
+          color: FButtonData.of(context).enabled ? _style.enabledSpinnerColor : _style.enabledSpinnerColor,
```

Comment:
Would it better to store the entire data instead? That way we can access both the data and the style without having to retrieve it each time.